llvm/test/CodeGen/DirectX/rootsignature-validation-textures-fail.ll

```diff
@@ -0,0 +1,16 @@
+; RUN: not opt -S -passes='dxil-post-optimization-validation' -mtriple=dxil-pc-shadermodel6.6-compute %s 2>&1 | FileCheck %s
+; CHECK: error: UAV at register 0 and space 0 is bound to a texture or typed buffer.
+
+@TB.str = private unnamed_addr constant [3 x i8] c"TB\00", align 1
+
+define void @CSMain() "hlsl.shader"="compute" {
+entry:
+  %TB =  tail call target("dx.Texture", float, 1, 0, 0, 4) @llvm.dx.resource.handlefrombinding.tdx.TypedBuffer_f32_1_0_0t(i32 0, i32 0, i32 1, i32 0, ptr nonnull @TB.str)
```

Comment:
You can skip the overload type here and let the IR parser add it implicitly
```suggestion
  %TB =  tail call target("dx.Texture", float, 1, 0, 0, 4) @llvm.dx.resource.handlefrombinding(i32 0, i32 0, i32 1, i32 0, ptr nonnull @TB.str)
```

Similarly in the other tests.